src/EventListener/DeserializerViewListener.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\EventListener;
+
+use ApiPlatform\Core\Api\RequestAttributesExtractor;
+use ApiPlatform\Core\Exception\RuntimeException;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\GetResponseForControllerResultEvent;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Updates the entity retrieved by the data provider with data contained in the request body.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+final class DeserializerViewListener
+{
+    private $serializer;
+
+    public function __construct(SerializerInterface $serializer)
+    {
+        $this->serializer = $serializer;
+    }
+
+    public function onKernelView(GetResponseForControllerResultEvent $event)
+    {
+        $data = $event->getControllerResult();
+        $request = $event->getRequest();
+        $content = $request->getContent();
+
+        if ($data instanceof Response || '' === $content) {
+            return;
+        }
+
+        try {
+            list($resourceClass, $collectionOperation, $itemOperation, $format) = RequestAttributesExtractor::extractAttributes($request);
+        } catch (RuntimeException $e) {
+            return;
+        }
+
+        $context = ['resource_class' => $resourceClass];
+        if ($collectionOperation) {
+            $context['collection_operation_name'] = $collectionOperation;
+        } else {
+            $context['item_operation_name'] = $itemOperation;
+        }
+
+        if (null !== $data) {
+            $context['object_to_populate'] = $data;
+        }
+
+        $event->setControllerResult($this->serializer->deserialize($content, $resourceClass, $format, $context));
```

Comment:
https://github.com/api-platform/core/blob/44a46e6d34c6c06d048ec7740d86515fec1e8282/src/EventListener/DeserializerViewListener.php#L40

(In the first implementation I was listing non-safe methods but this trick looks more extensible to me).